docs/content/guides/developer/app-examples/escrow-indexer-api.mdx

```diff
@@ -0,0 +1,634 @@
+---
+title: Escrow Indexing and API Service
+sidebar_label: Indexing and API Service
+---
+
+:::note Multi-Page Guide
+
+This is the second in a [three-part guide](/guides/developer/app-examples/escrow) on how to build a trustless atomic swap on Sui.
+
+:::
+
+In most cases where we want to enhance an dApp and ensure it is production ready, we need to have an indexing service (indexer) listening to the blockchain for on-chain data, shaping the data to fit our application needs, storing the transformed data into the local off-chain database so we can query them in the most efficient way. Joining hands with an indexer, we will expose an API allowing frontend to query the indexed data and update the screen as escrows are made and swaps are fulfilled.
+
+Architecturally, we will have the service doing the heavy-lifting indexing the data while the other service will expose the data through an API convention for external consumption.
+
+## Prerequisites
+
+:::info
+
+You can view the [complete source code for this app example](https://github.com/MystenLabs/sui/tree/main/examples/trading) in the Sui repository.
+
+:::
+
+Before getting started, make sure you:
+
+- Understand [the mechanism behind the Escrow smart contract backend](/guides/developer/app-examples/escrow).
+- Install [`pnpm` through this guide](https://pnpm.io/installation) as we will use it as our package manager.
+- Check out the [indexer's README](https://github.com/MystenLabs/sui/examples/trading/api) to setup the development environment.
+- Check out [Prisma](https://www.prisma.io/) to have an overall sense on what it is as we will use it to facilitate all the database interactions.
+- Check out [Express](https://expressjs.com/) to learn how to quickly setup a Web server application in Node.js as we will use it to bootstrap our API service.
+- Check out [Sui Typescript SDK](https://sdk.mystenlabs.com/typescript) for basic usage on how to interact with Sui with Typescript.
+
+## Indexing service
+
+### Data model
+
+In most cases, when we're working with a database directly from a backend service, we may want to use some sort of database libraries and toolings to abstract us away from database creation and management complexity. In our case, we use [Prisma](https://www.prisma.io/) to manage all the database interactions such as defining data models and database migrations.
```

Comment:
It's worth calling out the reason why we need indexing support -- it's to look-up `Escrow` objects by their `sender` and `recipient` (other queries can be handled over regular RPC).